src/lazy/encoder/binary/v1_1/flex_int.rs

```diff
@@ -66,6 +68,31 @@ impl FlexInt {
         Ok(FlexInt::new(flex_uint.size_in_bytes(), signed_value))
     }
 
+    // This is equivalent to calling `write_i64(my_bump_vec).unwrap()`, but optimized for writing
+    // to a `BumpVec` instead of a `W: Write`. Writing to a BumpVec cannot fail (barring out-of-
+    // memory errors and the like), which eliminates some branching, a loop inside
+    // `io::Write::write_all`, and the construction of a return value.
+    #[inline]
+    pub fn encode_i64(output: &mut BumpVec<u8>, value: i64) {
+        let encoded_size_in_bytes = if value < 0 {
+            BYTES_NEEDED_CACHE[value.leading_ones() as usize]
+        } else {
+            BYTES_NEEDED_CACHE[value.leading_zeros() as usize]
+        } as usize;
+        if encoded_size_in_bytes <= 8 {
+            // The entire encoding (including continuation bits) will fit in a u64.
+            // `encoded_size_in_bytes` is also the number of continuation bits we need to include
+            let mut encoded = value << encoded_size_in_bytes;
+            // Set the `end` flag to 1
+            encoded += 1 << (encoded_size_in_bytes - 1);
+            output.extend_from_slice_copy(&encoded.to_le_bytes()[..encoded_size_in_bytes]);
+            return;
+        }
+        cold_path! {{
+            let _ = Self::write_large_i64(output, value, encoded_size_in_bytes);
+        }}
```

Comment:
The downsides of marking the callee `#inline(never)` are:
1. That information isn't visible at the callsite, so a reader would need to go investigate it separately.
2. It prevents that method from being used in non-cold paths. (This isn't a problem for this method in particular since it's never going to be on the hot path, but it is an issue more generally.)
3. If any of the method's arguments are statically known at the call site, `cold_path!` bakes them into the resulting function so the assembly doesn't have to push or pop them from the call stack.
4. There may be other functionality that's part of the cold path but not part of the method. For example:

```rust
fn foo(&self) -> IonResult<()> {
    if self.something() {
        return Ok(());
    }
    cold_path! {{
        // This fn contains most of the cold path logic...
        fallible_fn()
        // ...but we need to map its result to an IonResult
            .map_err(IonError::from)
    }}
}
```